enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,494 @@
+
+Note: this is in line with [etcd-encryption enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md#proposal)
+with the exception that request level logging includes configmaps.
+
+Any kind of further, deeper policy configuration, e.g. that filters by API groups, user agents or namespaces 
+is explicitly not part of this proposal and requires work on the [upstream dynamic audit](https://github.com/kubernetes/enhancements/blob/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/sig-auth/0014-dynamic-audit-configuration.md)
+mechanism and its promotion to beta eventually.
+
+### User Stories [optional]
+
+#### Story 1
+
+As an average customer, I want to have a default audit policy which does not cause
+noticeable resource overhead (cpu and disk).
+
+#### Story 2
+
+As any customer, I don't want to leak sensible information of resources like secrets,
+oauth tokens, etc. in audit logs.
+
+#### Story 3
+
+As a security and regulator demanding customer, I want to log **every** non-sensible resource
+change up to request level detail. 
+
+#### Story 4
+
+As a security and regulator demanding customer, I want to log every non-sensible resource
+change up to request level detail **from non-OpenShift components**.
+
+#### Story 5
+
+As a security and regulator demanding customer, I want to log every non-sensible resource
+change up to request level detail, **from clients outside of the cluster (non-service-accounts)**.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The proposed API looks like this: 
+
+```yaml
+kind: APIServer
+apiVersion: config.openshift.io/v1
+spec:
+  ...
+  audit:
+    profile: Default | OauthUserRequests | NonOpenshiftRequests | AllRequests
+```
+
+In the future, we could add more, even more detailed profiles of logging responses, if customers need this:
+
+```yaml
+    profile: ... | OauthUserResponses | NonOpenShiftResponses | AllResponses
+```
+
+But we are not going to add these without a strong business case.
+
+### Risks and Mitigations
+
+- going beyond the proposed profiles, e.g. by filtering of API groups, user agents or namespaces, yields the risk that the
+  defined semantics do not map to the future [upstream dynamic audit](https://github.com/kubernetes/enhancements/blob/f1a799d5f4658ed29797c1fb9ceb7a4d0f538e93/keps/sig-auth/0014-dynamic-audit-configuration.md)
+  API types. In the file-based API of kube-apiserver, which implements powerful, but controversial rule behaviour, this kind
+  filters are possible, but it also turned out as the biggest blocker for moving it toward the dynamic variant as part of 
+  the Kubernetes REST API. We can not and must not follow this failure as it is big risk to the maintainability and future 
+  feasibility of this OpenShift API.
+
+## Design Details
+
+The profile defined in the `APIServer` singleton instance named `cluster` will be translated into the respective file-based audit policy of
```

Comment:
it's not just these.  To have fully audit power, you need every aggregated apiserver to read these.  Monitoring and service catalog (being removed) stand out.

Reply:
true. On the other hand, we cannot do the work for 3rdparty servers. That was the point here.